## Empirical Exercise: Birthweight and Smoking

In [1]:
# Install haven if not already installed, then load it
if (!require(haven)) install.packages("haven")
library(haven)

# Read the Birthweight_Smoking dataset
bw <- read_dta("Birthweight_Smoking.dta")

Loading required package: haven



In [2]:
# First few rows
head(bw)

nprevist,alcohol,tripre1,tripre2,tripre3,tripre0,birthweight,smoker,unmarried,educ,age,drinks
<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
12,0,1,0,0,0,4253,1,1,12,27,0
5,0,0,1,0,0,3459,0,0,16,24,0
12,0,1,0,0,0,2920,1,0,11,23,0
13,0,1,0,0,0,2600,0,0,17,28,0
9,0,1,0,0,0,3742,0,0,13,27,0
11,0,1,0,0,0,3420,0,0,16,33,0


In [3]:
# Summary statistics for all variables
summary(bw)

    nprevist        alcohol           tripre1         tripre2     
 Min.   : 0.00   Min.   :0.00000   Min.   :0.000   Min.   :0.000  
 1st Qu.: 9.00   1st Qu.:0.00000   1st Qu.:1.000   1st Qu.:0.000  
 Median :12.00   Median :0.00000   Median :1.000   Median :0.000  
 Mean   :10.99   Mean   :0.01933   Mean   :0.804   Mean   :0.153  
 3rd Qu.:13.00   3rd Qu.:0.00000   3rd Qu.:1.000   3rd Qu.:0.000  
 Max.   :35.00   Max.   :1.00000   Max.   :1.000   Max.   :1.000  
    tripre3         tripre0      birthweight       smoker        unmarried     
 Min.   :0.000   Min.   :0.00   Min.   : 425   Min.   :0.000   Min.   :0.0000  
 1st Qu.:0.000   1st Qu.:0.00   1st Qu.:3062   1st Qu.:0.000   1st Qu.:0.0000  
 Median :0.000   Median :0.00   Median :3420   Median :0.000   Median :0.0000  
 Mean   :0.033   Mean   :0.01   Mean   :3383   Mean   :0.194   Mean   :0.2267  
 3rd Qu.:0.000   3rd Qu.:0.00   3rd Qu.:3750   3rd Qu.:0.000   3rd Qu.:0.0000  
 Max.   :1.000   Max.   :1.00   Max.   :5755   Max.

a. *Regress Birthweight on Smoker. What is the estimated effect of smoking on birth weight?*

**[Ans]**

In [4]:
fit.bw <- lm(birthweight~smoker, data=bw)  #run the linear regression
summary(fit.bw)


Call:
lm(formula = birthweight ~ smoker, data = bw)

Residuals:
     Min       1Q   Median       3Q      Max 
-3007.06  -313.06    26.94   366.94  2322.94 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  3432.06      11.87 289.115   <2e-16 ***
smoker       -253.23      26.95  -9.396   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 583.7 on 2998 degrees of freedom
Multiple R-squared:  0.0286,	Adjusted R-squared:  0.02828 
F-statistic: 88.28 on 1 and 2998 DF,  p-value: < 2.2e-16


---

b. *Regress Birthweight on Smoker, Alcohol, and Nprevist.*

**[Ans]**

In [5]:
fit.bw2 <- lm(birthweight~smoker+alcohol+nprevist, data=bw)
summary(fit.bw2)


Call:
lm(formula = birthweight ~ smoker + alcohol + nprevist, data = bw)

Residuals:
     Min       1Q   Median       3Q      Max 
-2733.53  -307.57    21.42   358.09  2192.70 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 3051.249     34.016  89.701  < 2e-16 ***
smoker      -217.580     26.680  -8.155 5.07e-16 ***
alcohol      -30.491     76.234  -0.400    0.689    
nprevist      34.070      2.855  11.933  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 570.5 on 2996 degrees of freedom
Multiple R-squared:  0.07285,	Adjusted R-squared:  0.07192 
F-statistic: 78.47 on 3 and 2996 DF,  p-value: < 2.2e-16


---

i. *Using the two conditions in Key Concept 6.1, explain why the exclusion of *Alcohol* and *Nprevist* could lead to omitted variable bias in the regression estimated in (a).*

**[Ans]** Smoking may be correlated with both alcohol and the number of prenatal doctor visits, thus satisfying (1) in Key Concept 6.1. Moreover, both alcohol consumption and the number of doctor visits may have their own independent effects on birthweight, thus satisfying (2) in Key Concept 6.1.

---

ii. *Is the estimated effect of smoking on birth weight substantially different from the regression that excludes *Alcohol* and *Nprevist*? Does the regression in (a) seem to suffer from omitted variable bias?*

**[Ans]** The estimated effect of the multiple regression is smaller: it has fallen to 217 grams from 253 grams. Given that both conditions in Key Concept 6.1 are likely to be satisfied, the regression in (a) may suffer from omitted variable bias.

---

iii. *Jane smoked during her pregnancy, did not drink alcohol, and had 8 prenatal care visits. Use the regression to predict the birth weight of Jane's child.*

**[Ans]** The predicted birth weight of Jane's child is 3106.23.

In [6]:
coef.bw2 <- fit.bw2$coefficients
bw.pred.Jane <- coef.bw2[1] + coef.bw2[2]*1 + coef.bw2[3]*0 + coef.bw2[4]*8
print(bw.pred.Jane)

(Intercept) 
   3106.228 


---

iv. *Compute $R^2$ and $\bar{R}^2$. Why are they so similar?*

**[Ans]** $R^2 = 0.0729$, $\bar{R}^2 = 0.0719$. Their closeness arises from the large sample size ($n = 3000$), where the penalty term in $\bar{R}^2$, $(n-1)/(n-k-1)$, equals $(n-1)/(n-4) = 1.001$, making $\bar{R}^2$ almost identical to $R^2$.

---

v. *How should you interpret the coefficient on *Nprevist*? Does the coefficient measure a causal effect of prenatal visits on birth weight? If not, what does it measure?*

**[Ans]** *Nprevist* is a control variable. It captures, for example, mother's access to healthcare and health. Because *Nprevist* is a control variable, its coefficient does not have a causal interpretation.

---

**c.** *Estimate the coefficient on $\textit{Smoking}$ for the multiple regression model in (b), using the three-step process following the Frisch--Waugh theorem (See Appendix 6.3 in Stock and Whatson). Verify that the three-step process yields the same estimated coefficient for $\textit{Smoking}$ as that obtained in (b).*

In [7]:
# ── Frisch-Waugh Theorem: Three-Step Process ──────────────────────────────────

# Step 1: Regress Smoker on the other regressors (Alcohol, Nprevist)
#         and obtain residuals ~ "partialled-out" variation in Smoker
step1 <- lm(smoker ~ alcohol + nprevist, data = bw)
smoker_resid <- residuals(step1)

# Step 2: Regress Birthweight on the other regressors (Alcohol, Nprevist)
#         and obtain residuals ~ "partialled-out" variation in Birthweight
step2 <- lm(birthweight ~ alcohol + nprevist, data = bw)
bw_resid <- residuals(step2)

# Step 3: Regress the residuals from Step 2 on the residuals from Step 1
#         The coefficient is the Frisch-Waugh estimate of Smoker
step3 <- lm(bw_resid ~ smoker_resid)
round(coef(step3), 3)

# ── Verify: compare with the direct multiple regression ──────────────────────
fit_b <- lm(birthweight ~ smoker + alcohol + nprevist, data = bw)
round(coef(fit_b), 3)

# The coefficient on smoker should be identical in both

(Intercept) smoker_resid 
        0.00      -217.58

(Intercept)      smoker     alcohol    nprevist 
   3051.249    -217.580     -30.491      34.070

---

d. *An alternative way to control for prenatal visits is to use the binary variables Tripre0 through Tripre3. Regress Birthweight on Smoker, Alcohol, Tripre0, Tripre2, and Tripre3.*

In [8]:
fit.bw3 <- lm(birthweight~smoker+alcohol+tripre0+tripre2+tripre3, data=bw)
summary(fit.bw3)


Call:
lm(formula = birthweight ~ smoker + alcohol + tripre0 + tripre2 + 
    tripre3, data = bw)

Residuals:
     Min       1Q   Median       3Q      Max 
-3029.55  -307.55    31.35   372.45  2401.29 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  3454.55      12.65 273.077  < 2e-16 ***
smoker       -228.85      27.16  -8.424  < 2e-16 ***
alcohol       -15.10      77.54  -0.195 0.845613    
tripre0      -697.97     106.88  -6.531 7.66e-11 ***
tripre2      -100.84      29.62  -3.404 0.000672 ***
tripre3      -136.96      59.58  -2.299 0.021595 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 578.7 on 2994 degrees of freedom
Multiple R-squared:  0.04647,	Adjusted R-squared:  0.04487 
F-statistic: 29.18 on 5 and 2994 DF,  p-value: < 2.2e-16


---

i. *Why is Tripre1 excluded from the regression? What would happen if you included it in the regression?*

**[Ans]** Tripre1 is omitted to avoid perfect multicollinearity. (𝑇𝑟𝑖𝑝𝑟𝑒0+𝑇𝑟𝑖𝑝𝑟𝑒1+𝑇𝑟𝑖𝑝𝑟𝑒2+𝑇𝑟𝑖𝑝𝑟𝑒3=1, the value of the “constant” regressor that determines the intercept). If Tripre0, Tripre1, Tripre2, Tripre3, and the constant term all included in the regression, the software R will automatically drop one of the dummy variables and run the regression.

In [9]:
fit.bw4 <- lm(birthweight~smoker+alcohol+tripre0+tripre1+tripre2+tripre3, data=bw)
summary(fit.bw4)


Call:
lm(formula = birthweight ~ smoker + alcohol + tripre0 + tripre1 + 
    tripre2 + tripre3, data = bw)

Residuals:
     Min       1Q   Median       3Q      Max 
-3029.55  -307.55    31.35   372.45  2401.29 

Coefficients: (1 not defined because of singularities)
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  3317.59      59.00  56.231  < 2e-16 ***
smoker       -228.85      27.16  -8.424  < 2e-16 ***
alcohol       -15.10      77.54  -0.195   0.8456    
tripre0      -561.01     120.88  -4.641 3.61e-06 ***
tripre1       136.96      59.58   2.299   0.0216 *  
tripre2        36.12      64.17   0.563   0.5736    
tripre3           NA         NA      NA       NA    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 578.7 on 2994 degrees of freedom
Multiple R-squared:  0.04647,	Adjusted R-squared:  0.04487 
F-statistic: 29.18 on 5 and 2994 DF,  p-value: < 2.2e-16


---

ii. *The estimated coefficient on Tripre0 is large and negative. What does this coefficient measure? Interpret its value.*

**[Ans]** The estimated coefficient on Tripre0 is −697.97. It implies that babies born to women who had no prenatal doctor visits (𝑇𝑟𝑖𝑝𝑟𝑒0=1) had birth weights that on average were 697.97 grams (≈ 1.5 lbs) lower than babies from others who saw a doctor during the first trimester (𝑇𝑟𝑖𝑝𝑟𝑒1=1).

iii. *Interpret the value of the estimated coefficients on Tripre2 and Tripre3.*
**[Ans]** The estimated coefficients on Tripre2 and Tripre3 are −100.84 and −136.96, respectively. The results imply that babies born to women whose first doctor visit was during the second trimester (𝑇𝑟𝑖𝑝𝑟𝑒2=1) had birth weights that on average were 100.84 grams (≈0.2 lbs) lower than babies from others who saw a doctor during the first trimester (𝑇𝑟𝑖𝑝𝑟𝑒1=1). Babies born to women whose first doctor visit was during the third trimester (𝑇𝑟𝑖𝑝𝑟𝑒3=1) had birth weights that on average were 136.96 grams (≈0.3 lbs) lower than babies from others who saw a doctor during the first trimester (𝑇𝑟𝑖𝑝𝑟𝑒1=1).

iv. *Does the regression in (d) explain a larger fraction of the variance in birth weight than the regression in (b)?*
**[Ans]** The regression in (d) explains a smaller fraction of the variance in birth weight than the regression in (b) because 𝑅^2 in (d) is smaller than 𝑅^2 in (b).